File: src/shared/Mover.Stats.Shared/Models/Activity.cs
using System.Text.Json.Serialization;

namespace Mover.Stats.Shared.Models;
public class Activity
{
    [JsonPropertyName("idRaw")]
    public string IdRaw { get; set; }

    [JsonPropertyName("userId")]
    public string UserId { get; set; }

    [JsonPropertyName("fn")]
    public string FileName { get; set; }

    [JsonPropertyName("activityType")]
    public string ActivityType { get; set; }

    [JsonPropertyName("timestamp")]
    public long Timestamp { get; set; }

    [JsonPropertyName("tTime")]
    public long TotalTime { get; set; }

    [JsonPropertyName("tPDrop")]
    public double TotalPositiveDrop { get; set; }

    [JsonPropertyName("tDistance")]
    public double TotalDistance { get; set; }

    [JsonPropertyName("points")]
    public IEnumerable<Point> Points { get; set; }
}

File: src/frontend/mover/src/models/activity.ts
export type activity = {
    idRaw: string;
    activityType: string;
    timestamp: number;
    tTime: number;
    tPDrop: number;
    tDistance: number;
};

export type activityFull = activity & {
    fn: string;
    points: point[];
    positions: position[];
};

export type point = {
    ts: number;
    temp: number;
    ele: number;
    s: number;
    hr: number;
    c: number;
};

export type position = {
    lat: number;
    lng: number;
};
